docs/src/references/child-devices.md

```diff
@@ -0,0 +1,45 @@
+# Child Device Concept (Solution Proposal)
+
+Thin-edge provides a child-device concept to 
+TODO: Add motivation for child-device concept. Somethin like:
+-- Device-side and Cloud-side structural view (that might match the physical structure)
+-- Features are available per child-device (e.g. Software Management with one SW list per child device, ...)
+
+* Child-Device support is provided on thin-edge public interfaces as below:
+    - tedge/measurements
+    - tedge/alarms
+    - tedge/events
+    - SW Management plugin API
+    - Configuration Management
+    - Logging Management
+    
+* Each public interface that provides child-device support has an optional field "child-device ID".
+
+* Interfaces available on MQTT (as measurements, alarms, events) manage the field "child-device ID" as last topic of a message.
+  Examples:
+  
+  ```
+     tedge/measurements/<child-device ID>
+     tedge/events/<event-type>/<child-device ID>
+  ```
+
+- Interface available via CLI (as SW Management Plugin API) manage the field "child-device ID" as optional argument in the command call.
+  That means the command is executed locally on the thin-edge device, and has to take care to execute the remote action on the child-device.
```

Comment:
Providing software management to child devices via SW management plugin APIs(CLIs) limits the solution to only the cases where the thin-edge daemons and child devices processes are running on the same host with a shared file system. For the use-cases where thin-edge itself is running as a container and the child device process is running separately from it (either as a native process or another container), this CLI approach won't work as the tedge-agent in the thin-edge container won't be able to invoke the CLI that's installed natively on the device or in another container.

Reply:
That section shall not mean we keep sticking with the SW Management Plugin API on CLI.

Instead I want to define for all public interface-shapes of thin-edge a general way how handle child-devices. 

And for that I see in general two aspects to address:
1) allow to associate a certain item on the device side with a child device-twin in the cloud
2) allow to provide/consume a certain item to/from an external device / container

NOTE: Such an item could be a measurement, an alarm, a config file of C8Y Cfg Plugin, ...

You are right, that _child-device ID_ in the CLI just solves aspect (1). But that can be valid if the child-device is represented by a local process on the thin-edge device (e.g. an OT protocol stack). And if the child-device is instead represented by an external device or container, then CLI is simply the wrong interface to handle that.

In the end to solve aspect (2) from above, we have to decide for all public interfaces (especially for those that come over CLI) whether we want to heave/replace them with some more portable one (e.g. some network protocol (MQTT?)).